### PR TITLE
Introduce new method to remove non-alphanumeric characters from a string

### PR DESCRIPTION
**Pull request by Norbinsh**
Hey.

this PR addresses #18.

Happy to circle back on changes based on feedback.

Thank you.

### CODE REVIEW

**Comment by Norbinsh** [comment]
Looks like the CI partially fail due to the below:

"Please provide an upload token from codecov.io with valid arguments"

File: regex/rstring/string.go [hunk 1]
```diff
@@ -41,3 +42,16 @@ func SplitString(n int, strs ...string) []string {
 
 	return split
 }
+
+// RemoveNonAlphaNumeric will remove any non-alphanumeric character from the input string
+// And return a newly processed string.
+func RemoveNonAlphaNumeric(input string) string {
+	pattern := "[^a-zA-Z0-9]+"
```
Comment (patil-ashutosh) [inline regex/rstring/string.go:49]: @Norbinsh do we really need '+' in pattern ("[^a-zA-Z0-9]+"),  if we are replacing string

**Comment by patil-ashutosh** [comment]
> Looks like the CI partially fail due to the below:
> 
> "Please provide an upload token from codecov.io with valid arguments"

@Norbinsh please address comments and try again , i fixed codecov token issue.
please refresh branch

File: regex/rstring/string.go (hunk 1 above)
Comment (Norbinsh) [reply regex/rstring/string.go]: * updated my fork with latest master
* removed repeating expression (plus sign) as it's not required - thanks.

**Comment by Norbinsh** [comment]
> > Looks like the CI partially fail due to the below:
> > "Please provide an upload token from codecov.io with valid arguments"
> 
> @Norbinsh please address comments and try again , i fixed codecov token issue.
> please refresh branch

please let me know if there's anything to be changed still

**Review by patil-ashutosh — APPROVED** [review]

**Comment by patil-ashutosh** [comment]
@Norbinsh  Thanks for working on this. You can also work on open new issue/feature relevant to this repo

**Pull request merged**